### ISSUE TITLE
Wishlist: Default to app/remote that current branch is tracking

### ISSUE THREAD

**Issue by joallard** [heroku/heroku#1874]
Building on #277, I also have multiple environments. Say I'm currently on the `develop` branch that is tracking the `staging` remote at `master`. I could expect that all `heroku` commands run parallel to `git push`, so that `-r staging` is implied if I'm working on the `develop` branch and so on. That way, I wouldn't need to type the `-r` every time I run `config:set` or `pg:psql`, `logs`, etc.

I'm conscious this might not be the ideal behavior for everyone. Were one not conscious of this, it could be easy to overwrite settings involuntarily on a production app for instance. Adding a setting about that behavior sounds like the right way to go.

**Comment by dickeyxxx** [comment]
You could do this with a git post-checkout hook: (create `.git/hooks/post-checkout` with execute permissions)

```
#!/bin/bash

BRANCH=`git rev-parse --abbrev-ref HEAD`
echo "changing heroku remote to $BRANCH"
git config heroku.remote $BRANCH
```

**Comment by dickeyxxx** [comment]
This couldn't be default functionality, too much change from how it works today, but maybe something configurable.

**Comment by joallard** [comment]
I agree. A configurable option would be pretty cool I think. Thanks for the workaround, hadn't thought of that.

**Comment by joallard** [comment]
By the way, as someone whose branch names don't equal the remote names (`develop: staging, master: production`), here's what I was angling for:

```
#!/bin/bash

BRANCH=$(git rev-parse --abbrev-ref HEAD)
REMOTE=$(git config branch.$BRANCH.remote)
echo "post-checkout: Changing heroku.remote to \`$REMOTE\`"
git config heroku.remote $REMOTE
```

And I'm as happy as a fish in the water.
